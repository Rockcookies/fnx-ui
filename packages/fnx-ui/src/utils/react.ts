import { Children, ReactElement } from 'react';
import { isFragment } from 'react-is';

export const toElementArray = (
	children: React.ReactNode,
	option?: { keepEmpty?: boolean },
): ReactElement[] => {
	const keepEmpty = option && option.keepEmpty;
	let ret: React.ReactElement[] = [];

	Children.forEach(children, (child: any) => {
		if ((child === undefined || child === null) && !keepEmpty) {
			return;
		}

		if (Array.isArray(child)) {
			ret = ret.concat(toElementArray(child));
		} else if (isFragment(child) && child.props) {
			ret = ret.concat(toElementArray(child.props.children, option));
		} else {
			ret.push(child);
		}
	});

	return ret;
};
