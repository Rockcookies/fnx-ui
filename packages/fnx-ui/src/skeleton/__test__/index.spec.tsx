import React from 'react';
import { render } from '@testing-library/react';
import Skeleton from '../index';
import TestsDOM from '../../../test/dom';

describe('<Skeleton/>', () => {
	it('should render with row width array correctly', async () => {
		const { container } = render(
			<Skeleton rowWidth={['100%', 30, '5rem']} row={3} loading={true} />,
		);

		expect(container.querySelectorAll('.fnx-skeleton__row').length).toBe(3);
	});

	it('should render default slot when loading is false', async () => {
		const { container } = render(
			<Skeleton loading={false}>
				<div className="foo">Content</div>
			</Skeleton>,
		);

		expect(container.querySelector('.foo')).not.toBeNull();
	});

	it('should change avatar size when using avatar-size prop', async () => {
		const { container } = render(
			<Skeleton data-testid="skeleton" avatar={true} avatarSize="200px">
				<div className="foo">Content</div>
			</Skeleton>,
		);

		const track = TestsDOM.mustQuerySelector<HTMLElement>(
			container,
			'.fnx-skeleton__avatar',
		);

		expect(track.style.width).toMatchSnapshot('200px');
	});

	it('should change avatar shape when using avatar-shape prop', async () => {
		const { getByTestId } = render(
			<Skeleton
				data-testid="skeleton"
				avatar={true}
				avatarShape={'square'}
			>
				<div className="foo">Content</div>
			</Skeleton>,
		);

		expect(getByTestId('skeleton')).toMatchSnapshot();
	});

	it('should be round when using round prop', async () => {
		const { container } = render(
			<Skeleton
				data-testid="skeleton"
				avatar={true}
				title={true}
				row={3}
				round={true}
			>
				<div className="foo">Content</div>
			</Skeleton>,
		);

		const track = TestsDOM.mustQuerySelector(
			container,
			'.fnx-skeleton--round',
		);

		expect(track).not.toBeNull();
	});

	it('should allow to disable animation', async () => {
		const { container } = render(
			<>
				<Skeleton title={true} animate={true}></Skeleton>,
				<Skeleton title={true} animate={false}></Skeleton>,
			</>,
		);

		expect(container).toMatchSnapshot();
	});
});
