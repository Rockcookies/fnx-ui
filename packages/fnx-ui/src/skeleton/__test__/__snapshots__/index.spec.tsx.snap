// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton/> should allow to disable animation 1`] = `
<div>
  <div
    class="fnx-skeleton fnx-skeleton--animate"
  >
    <div
      class="fnx-skeleton__content"
    >
      <h3
        class="fnx-skeleton__title"
      />
      <div
        class="fnx-skeleton__row"
        style="width: 100%;"
      />
      <div
        class="fnx-skeleton__row"
        style="width: 100%;"
      />
      <div
        class="fnx-skeleton__row"
        style="width: 60%;"
      />
    </div>
  </div>
  ,
  <div
    class="fnx-skeleton"
  >
    <div
      class="fnx-skeleton__content"
    >
      <h3
        class="fnx-skeleton__title"
      />
      <div
        class="fnx-skeleton__row"
        style="width: 100%;"
      />
      <div
        class="fnx-skeleton__row"
        style="width: 100%;"
      />
      <div
        class="fnx-skeleton__row"
        style="width: 60%;"
      />
    </div>
  </div>
  ,
</div>
`;

exports[`<Skeleton/> should change avatar shape when using avatar-shape prop 1`] = `
<div
  class="fnx-skeleton fnx-skeleton--animate"
  data-testid="skeleton"
>
  <div
    class="fnx-skeleton__avatar fnx-skeleton__avatar--square"
  />
  <div
    class="fnx-skeleton__content"
  >
    <div
      class="fnx-skeleton__row"
      style="width: 100%;"
    />
    <div
      class="fnx-skeleton__row"
      style="width: 100%;"
    />
    <div
      class="fnx-skeleton__row"
      style="width: 60%;"
    />
  </div>
</div>
`;

exports[`<Skeleton/> should change avatar size when using avatar-size prop: 200px 1`] = `"200px"`;
